docs/development/typescript.md

```diff
@@ -0,0 +1,64 @@
+# Typescript
+
+**Question**: should we recommend using typescript over javascript?
+
+**TODO:** advantages of typescript over javascript?
+
+Node.js does not support running typescript natively, so it must first be transpiled to JavaScript. There are 2 widely used transpilers:
+
+1. `tsc` - the compiler shipped with [typescript](https://www.npmjs.com/package/typescript), supports type checking and transpiles to JavaScript
```

Comment:
This got me thinking. Does it make sense to separate this document in to client and server sections? On the server-side I don't think there's much use for Babel (unless you *really* want to use some new spec/polyfills?) since you know and control the JS runtime. 

Should we mention it's possible, but have more strongly recommend `tsc`?